tests/crashes/137187.rs

```diff
@@ -1,9 +1,13 @@
 //@ known-bug: #137187
+#![feature(const_trait_impl, const_ops)]
+
 use std::ops::Add;
-trait A where
-    *const Self: Add,
+const trait A where
+    *const Self: const Add,
 {
-    const fn b(c: *const Self) -> <*const Self as Add>::Output {
+    fn b(c: *const Self) -> <*const Self as Add>::Output {
         c + c
     }
 }
+
+fn main() {}
```

Comment:
This produces the same ICE as before, but the changes were required to pass typeck so we'd actually get to mir const checks